docs/extensibility.md

```diff
@@ -29,531 +28,630 @@ pub trait DiscoveryHandler {
+        - name: SECRET
+      containers:
+        - name: device
+          image: IMAGE
+          imagePullPolicy: Always
+          args:
+            - --path=/
+            - --device=http://device-1:8080
+            - --device=http://device-2:8080
+            - --device=http://device-3:8080
+            - --device=http://device-4:8080
+            - --device=http://device-5:8080
+            - --device=http://device-6:8080
+            - --device=http://device-7:8080
+            - --device=http://device-8:8080
+            - --device=http://device-9:8080
+          ports:
+            - name: http
+              containerPort: 8080
+```
 
-```sh
-# Build and push ghcr.io/<GITHUB USERNAME>/agent:nessie to container repo
-LABEL_PREFIX=extensibility PREFIX=ghcr.io/<GITHUB USERNAME> BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-agent
-# Build and push ghcr.io/<GITHUB USERNAME>/controller:nessie to container repo
-LABEL_PREFIX=extensibility PREFIX=ghcr.io/<GITHUB USERNAME> BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-controller
+Then apply `device.yaml` to create a Deployment (called `device`) and a Pod (called `device-...`):
+
+```bash
+kubectl apply --filename=./samples/apps/http-apps/kubernetes/device.yaml
+```
+
+> **NOTE** We're using one Deployment|Pod to represent 9 devices AND a discovery service ... we will create 9 (distinct) Services against it (1 for each mock device) and 1 Service to present the disvoery service.
+
+Then create 9 mock device Services:
+
+```bash
+for NUM in {1..9}
+do
+  # Services are uniquely named
+  # The service uses the Pods port: 8080
+  kubectl expose deployment/device \
+  --name=device-${NUM} \
+  --port=8080 \
+  --target-port=8080 \
+  --labels=id=akri-http-device
+done
+```
+
+> Optional: check one the services:
+>
+> ```bash
+> kubectl run curl -it --rm --image=curlimages/curl -- sh
+> ```
+>
+> Then, pick a value for `X` between 1 and 9:
+>
+> ```bash
+> X=6
+> curl device-${X}:8080
+> ```
+>
+> Any or all of these should return a (random) 'sensor' value.
+
+Then create a Service (called `discovery`) using the deployment:
+
+```bash
+kubectl expose deployment/device \
+--name=discovery \
+--port=8080 \
+--target-port=8080 \
+--labels=id=akri-http-device
+```
+
+> Optional: check the service to confirm that it reports a list of devices correctly using:
+> 
+> ```bash
+> kubectl run curl -it --rm --image=curlimages/curl -- sh
+> ```
+>
+> Then, curl the service's endpoint:
+>
+> ```bash
+> curl discovery:8080/discovery
+> ```
+>
+> This should return a list of 9 devices, of the form `http://device-X:8080`
+
+
+# Where the rubber meets the road!
+At this point, we've extended Akri to include discovery for our HTTP protocol and we've created an HTTP broker that can be deployed.  Let's take HTTP for a spin!!
+
+## Deploy Akri
+
+> Optional: If you've previous installed Akri and wish to reset, you may:
+>
+> ```bash
+> # Delete Akri Helm
+> sudo microk8s.helm3 uninstall akri

```

Comment:
this isn't microk8s specific right? Should we remove all microks8 (and helm3) specific references?